vms/avm/states/state.go

```diff
@@ -576,3 +578,50 @@ func (s *state) writeStatuses() error {
 	}
 	return nil
 }
+
+func (s *state) rejectedTxs() ([]ids.ID, error) {
+	var txIDs []ids.ID
+	iter := s.statusDB.NewIterator()
+	for iter.Next() {
+		key := iter.Key()
+		val := iter.Value()
+		valStatus, err := database.ParseUInt32(val)
+		if err != nil {
+			return []ids.ID{}, err
+		}
+		if choices.Status(valStatus) == choices.Rejected {
+			txID, err := ids.ToID(key)
+			if err != nil {
+				return []ids.ID{}, err
+			}
+			txIDs = append(txIDs, txID)
+			// remove txID from dbStatus
+			if err := database.ClearPrefix(s.statusDB, s.statusDB, key); err != nil {
+				return []ids.ID{}, err
+			}
+		}
+	}
+	return txIDs, nil
+}
+
+func (s *state) RemoveRejectedTxs() error {
+	txIDs, err := s.rejectedTxs()
+	if err != nil {
+		return err
+	}
+	for _, txID := range txIDs {
+		tx, err := s.GetTx(txID)
+		if err != nil {
+			return err
+		}
+		// find UTXOID produced by the tx
+		utxos := tx.UTXOs()
```

Comment:
We should not be touching produced UTXOs.

Reply:
updated